sorting/quick_sort_iterative.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * @file
+ * @brief Quick Sort without recursion. This method uses the stack instead.
+ * Both recursive and iterative implementations have O(n log n) best case
+ * and O(n^2) worst case.
+ * @details
+ * https://stackoverflow.com/questions/12553238/quicksort-iterative-or-recursive
+ * https://en.wikipedia.org/wiki/Quicksort
+ * https://www.geeksforgeeks.org/iterative-quick-sort/
+ * @author [Sebe324](https://github.com/sebe324)
+ */
+
+#include <iostream> /// for std::cout
+#include <vector> /// for std::vector
+#include <stack> /// for std::stack
+#include <algorithm> /// for std::is_sorted
+#include <cassert> /// for assert
+
```

Comment:
```suggestion


/**
 * @namespace sorting
 * @brief Sorting algorithms
 */
namespace sorting {
```
should be inside sorting namespace